app/models/bookings/schools_subject.rb

```diff
@@ -0,0 +1,18 @@
+class Bookings::SchoolsSubject < ApplicationRecord
+  belongs_to :bookings_school,
+    class_name: "Bookings::School",
+    optional: true
```

Comment:
Is this to avoid the presence check? 

If so isn't that only a problem because it doesn't know the inverse relationship?

Logically I can't see a scenario where we want a join record which doesn't join to anything?

Reply:
I think the underlying issue on this is the relationship isn't fully defined and rails is failing to infer it - wondering whether the addition of `inverse_of` or similar would solve the mutual dependency issue.